### PR TITLE
[TASK-254] agent login all queues if no previous logged in queue

### PR DESCRIPTION
**Pull request by bloom1**
> [!NOTE]
> **Medium Risk**
> Changes agent login flow to implicitly log into all assigned queues when no prior queue membership state exists, which can affect call routing and queue membership behavior. Risk is moderate due to new side effects during login and reliance on `QueueManager`/bus events, but scoped to this edge case and covered by new tests.
> 
> **Overview**
> When an agent logs in and has **no previously enabled/logged queues**, the login flow now automatically logs the agent into *all queues assigned to them* (via `QueueManager.login_to_queue`), emitting the usual queue login bus events.
> 
> Wires `QueueManager` into `LoginAction` and adjusts startup wiring in `main.py`, and updates `_Queue` to reuse the DAO-provided type instead of a local `namedtuple`.
> 
> Adds unit + integration coverage for the new “no previous queue state” behavior and for the case where an agent has no queues assigned.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit f6b6ae2b752d17acaf864bae7d333948592f4e97. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: wazo_agentd/service/action/login.py [hunk 1]
```diff
@@ -71,10 +72,32 @@ def login_agent_on_line(self, agent, line_id):
     def _do_login(self, agent, extension, context, interface, state_interface):
         self._update_agent_status(agent, extension, context, interface, state_interface)
         self._update_queue_log(agent, extension, context)
+        self._ensure_queues_logged_in(agent)
```
Comment (ja0x) [inline wazo_agentd/service/action/login.py]: Not a blocker; logically speaking, we are sending bus events *before* the actual changes happen in Asterisk, which is a smell. Not an issue because AMI failures are swallowed, which doesnt fix the outcome.
A real conversation will be needed to harden Agentd AMI failures for correctness

File: wazo_agentd/service/manager/queue.py [hunk 2]
```diff
@@ -107,3 +113,12 @@ def logoff_from_queue(self, agent: Agent, queue: Queue) -> None:
...
             self._send_bus_event(UserAgentQueueLoggedOffEvent, agent, agent_queue)
+
+            with db_utils.session_scope():
+                agent_status = self._agent_status_dao.get_status(agent.id)
+
+            if agent_status and not any(q.logged for q in agent_status.queues):
+                logger.info(
+                    'Agent %s has no remaining logged queues, logging off', agent.id
+                )
+                self._logoff_action.logoff_agent(agent_status)
```
Comment (ja0x) [inline wazo_agentd/service/manager/queue.py]: Didn't check but can this raise? Might be worth wrapping and reformat the message if needed

**Review by ja0x — CHANGES_REQUESTED** [review]
LGTM, just curious about your take on the logoff_agent
We'll need to make sure we document the behavior change somewhere

File: wazo_agentd/service/action/login.py (hunk 1 above)
Comment (bloom1) [reply wazo_agentd/service/action/login.py]: Yes, I'm not sure it's a good idea to send those events in the first place... They are also sent before the agent login event is sent. I'll rework that a bit

File: wazo_agentd/service/manager/queue.py (hunk 2 above)
Comment (bloom1) [reply wazo_agentd/service/manager/queue.py]: Sorry, that was a commit not supposed to be there. It's from another PR. I removed it.

File: wazo_agentd/service/action/login.py [hunk 3]
```diff
@@ -73,8 +77,30 @@ def _do_login(self, agent, extension, context, interface, state_interface):
...
+        if enabled_queues or not agent.queues:
+            return
+        logger.info(
+            'Agent %s has no previously logged queues, '
+            'logging into all assigned queues',
+            agent.id,
+        )
+        for queue in agent.queues:
+            try:
+                self._queue_manager.login_to_queue(agent, queue)
```
[the remark this answers is not in the document]
Comment (bloom1) [reply wazo_agentd/service/action/login.py:95]: Nope, in xivo-dao we see that it uses the right _Queue and _Agent wrappers.
See https://github.com/wazo-platform/xivo-dao/blob/master/xivo_dao/agent_dao.py#L102

[the remark this answers is not in the document]
Comment (ja0x) [reply wazo_agentd/service/action/login.py:95]: They are different but equilavent nametuples, safe to use, but definitely a smell

**Review by ja0x — APPROVED** [review]
